pkg/rpcbackend/backend.go

```diff
@@ -127,6 +145,17 @@ func (rc *RPCClient) CallRPC(ctx context.Context, result interface{}, method str
 // In all return paths *including error paths* the RPCResponse is populated
 // so the caller has an RPC structure to send back to the front-end caller.
 func (rc *RPCClient) SyncRequest(ctx context.Context, rpcReq *RPCRequest) (rpcRes *RPCResponse, err error) {
+	if rc.concurrencySlots != nil {
+		select {
+		case rc.concurrencySlots <- true:
+			// wait for the concurrency slot and continue
+		case <-ctx.Done():
+			return nil, fmt.Errorf("request with id %s failed due to canceled context", rpcReq.ID)
```

Comment:
Sorry @Chengxuan, all errors should be translated in FF repos

Think you need to move this here: https://github.com/hyperledger/firefly-signer/blob/main/internal/signermsgs/en_error_messges.go